qc-runner/src/dataIntegration.r

```diff
@@ -99,31 +104,36 @@ run_dataIntegration <- function(scdata, config){
     # default for FindIntegrationAnchors
     k.filter <- 200 
 
+    # By defatult, we need RNA assay to compute the integrated matrix.
+    Seurat::DefaultAssay(scdata) <- "RNA"
+
+    # @misc slots not preserved so transfer
+    misc <- scdata@misc
+
     # temporary to make sure we don't run integration if unisample
     nsamples <- length(unique(scdata$samples))
-    if(nsamples>1 && method!="noIntegration"){
+    if(nsamples>1 && method!="unisample"){
         if(method=="seuratv4"){
             # Currently, we only support Seurat V4 pipeline for the multisample integration
-            #FIX FOR CURRENT DATASET!!!!!!
-            Seurat::DefaultAssay(scdata) <- "RNA"
             data.split <- Seurat::SplitObject(scdata, split.by = "samples")
             for (i in 1:length(data.split)) {
                 data.split[[i]] <- Seurat::NormalizeData(data.split[[i]], normalization.method = normalization, verbose = F)
                 data.split[[i]] <- Seurat::FindVariableFeatures(data.split[[i]], selection.method = "vst", nfeatures = nfeatures, verbose = FALSE)
             }
+
+            # The active reduction in seuratv4 is pca
+            active.reduction <- "pca"
+
             # If Number of anchor cells is less than k.filter/2, there is likely to be an error:
             # Note that this is a heuristic and was found to still fail for small data-sets
 
             # Try to integrate data (catch error most likely caused by too few cells)
-
             tryCatch({
             k.filter <- min(ceiling(sapply(data.split, ncol)/2), k.filter)
             data.anchors <- Seurat::FindIntegrationAnchors(object.list = data.split, dims = 1:numPCs, k.filter = k.filter, verbose = TRUE)
 
-            # @misc slots not preserved so transfer
-            misc <- scdata@misc
             scdata <- Seurat::IntegrateData(anchorset = data.anchors, dims = 1:numPCs)
-            scdata@misc <- misc
```

Comment:
This can't be removed!!! It led to a very difficult to debug bug that Alex managed to solve with this magic line

Reply:
Since we have the same problem for FastMNN, I have included this line before (line 11).